login/login.go

```diff
@@ -0,0 +1,96 @@
+package login
+
+import (
+	"bufio"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/juju/cmd"
+	"github.com/juju/usso"
+	"golang.org/x/crypto/ssh/terminal"
+	"gopkg.in/errgo.v1"
+)
+
+type TokenGetter interface {
+	GetTokenWithOTP(email, password, otp, tokenName string) (*usso.SSOData, error)
+}
+
+var ussoServer (TokenGetter) = usso.ProductionUbuntuSSOServer
+
+// LoginUSSO reads login parameters from ctxt.Stdin and then retrieves an
+// oauth token from Ubuntu SSO.
+func LoginUSSO(ctx *cmd.Context, twoFactor bool, tokenPath string) (*usso.SSOData, error) {
```

Comment:
I don't like the tokenPath parameter here, possibly it would be better as.
```
func LoginUSSO(ctx *cmd.Context, twoFactor bool, save func(*usso.SSOData)error) (*usso.SSOData, error) {
...
```
then if save != nil call it to save the data?